tests/test_butlerUtils.py

```diff
@@ -0,0 +1,394 @@
+
+    def test_LATISS_REPO_LOCATION_MAP(self):
+        self.assertTrue(LATISS_REPO_LOCATION_MAP is not None)
+        self.assertTrue(LATISS_REPO_LOCATION_MAP != [])
+        self.assertTrue(len(LATISS_REPO_LOCATION_MAP) >= 1)
+        self.assertTrue(len(LATISS_REPO_LOCATION_MAP) >= len(LATISS_SUPPLEMENTAL_COLLECTIONS))
+
+    def test_LATISS_DEFAULT_COLLECTIONS(self):
+        self.assertTrue(LATISS_DEFAULT_COLLECTIONS is not None)
+        self.assertTrue(LATISS_DEFAULT_COLLECTIONS != [])
+        self.assertTrue(len(LATISS_DEFAULT_COLLECTIONS) >= 1)
+
+    def test_RECENT_DAY(self):
+        todayInt = int(datetime.date.today().strftime("%Y%m%d"))
+        self.assertTrue(RECENT_DAY <= todayInt)  # in the past
+        self.assertTrue(RECENT_DAY >= 20200101)  # not too far in the past
+
+        # no test here, but print a warning if it hasn't been updated recently
+        recentDay_datetime = datetime.datetime.strptime(str(RECENT_DAY), "%Y%m%d")
+        now = datetime.datetime.today()
+        timeSinceUpdate = now - recentDay_datetime
+        if timeSinceUpdate.days > 100:  # TODO:
+            print(f"RECENT_DAY is now {timeSinceUpdate.days} days in the past. "
+                  "You might want to consider updating this to speed up butler queries.")
+
+    def test_sanitize_day_obs(self):
+        dayObs = '2020-01-02'
+        self.assertEqual(sanitize_day_obs(dayObs), 20200102)
+        dayObs = 20210201
+        self.assertEqual(sanitize_day_obs(dayObs), dayObs)
+
+        with self.assertRaises(ValueError):
+            sanitize_day_obs(1.234)
+            sanitize_day_obs('Febuary 29th, 1970')
+
+    def test_getMostRecentDayObs(self):
+        # just a basic sanity check here as we can't know the value,
+        # but at least check something is returned, and is plausible
+        recentDay = getMostRecentDayObs(self.butler)
+        self.assertTrue(isinstance(recentDay, int))
+        self.assertTrue(recentDay >= RECENT_DAY)
+        # some test data might be set a millennium in the future, i.e.
+        # the year wouldd be 2XXX+1000, so set to y4k just in case
+        self.assertTrue(recentDay < 40000000)
+
+    def test_getSeqNumsForDayObs(self):
+        emptyDay = 19990101
+        seqnums = getSeqNumsForDayObs(self.butler, emptyDay)
+        self.assertTrue(isinstance(seqnums, Iterable))
+        self.assertEqual(len(list(seqnums)), 0)
+
+        recentDay = getMostRecentDayObs(self.butler)
+        seqnums = getSeqNumsForDayObs(self.butler, recentDay)
+        self.assertTrue(isinstance(seqnums, Iterable))
+        self.assertTrue(len(list(seqnums)) >= 1)
+
+    def test_getMostRecentDataId(self):
+        # we can't know the values, but it should always return something
+        # and the dict and int forms should always have certain keys and agree
+        dataId = getMostRecentDataId(self.butler)
+        self.assertTrue(isinstance(dataId, dict))
+        self.assertTrue('day_obs' in dataId)
```

Comment:
Preferable to use `self.assertIn("day_obs", dataId)` because you get a much better error message when it isn't there (False is not True!)